### PR TITLE
append all given options to lp command

### PR DESCRIPTION
**Pull request by dVelopment**
this PR allows to add all options accepted with `lp -o OPTION[=VALUE]`. f.e.

```js
printer.print(pdf, {
    printer: 'Zebra',
    'fit-to-page': true,
    media: 'A4',
    landscape: true
});
```

will result in `lp -d Zebra -o fit-to-page -o media=A4 -o landscape`

### CODE REVIEW

**Comment by artiebits** [comment]
Hi Daniel, thank you very much for contributing to this project. 

I believe if we add any options to pdf-to-printer API they should work on both Unix-like and Windows platforms. Can you take a look at SumatraPDF docs and find equivalents to `fit-to-page`, `media` and `landscape`, please?

I also believe we should add all these new options to README.md otherwise nobody will know they are available.

**Comment by FrancescoBorzi** [comment]
@dVelopment  please check the build, it is failing

File: src/unix/print.js [hunk 1]
```diff
@@ -8,6 +8,18 @@ const getCommand = (pdf, options) => {
   if (options.printer) {
     command += " -d " + options.printer;
   }
+  // check for other options
+  Object.keys(options).forEach(function(key) {
+    if (key !== "printer") {
```
Comment (FrancescoBorzi) [inline src/unix/print.js]: please use single quotes

File: src/unix/print.js [hunk 2]
```diff
@@ -8,6 +8,18 @@ const getCommand = (pdf, options) => {
...
+      const value = options[key];
+      if (value !== false) {
+        command += " -o " + key;
```
Comment (FrancescoBorzi) [inline src/unix/print.js]: please use single quotes

File: src/unix/print.js [hunk 3]
```diff
@@ -8,6 +8,18 @@ const getCommand = (pdf, options) => {
   if (options.printer) {
     command += " -d " + options.printer;
   }
+  // check for other options
+  Object.keys(options).forEach(function(key) {
+    if (key !== "printer") {
+      const value = options[key];
+      if (value !== false) {
```
Comment (FrancescoBorzi) [inline src/unix/print.js]: why not just `if (!!value) {`

File: src/unix/print.js [hunk 4]
```diff
@@ -8,6 +8,18 @@ const getCommand = (pdf, options) => {
...
+        command += " -o " + key;
+        if (typeof value !== "boolean") {
+          command += "=" + value;
```
Comment (FrancescoBorzi) [inline src/unix/print.js]: what about ```command = `${command} = ${value}`;```

File: src/unix/print.js (hunk 3 above)
Comment (dVelopment) [reply src/unix/print.js]: because f.e. `{ 'some-option': 0 }` would not be serialized properly with `!! value`

File: src/unix/print.js (hunk 4 above)
Comment (dVelopment) [reply src/unix/print.js]: sure, no problem. Just wanted to keep the convention I found in that file ...

**Comment by dVelopment** [comment]
@artiebits I'll gladly put the example from my tests into the docs and link to [Command-Line Printing and Options](https://www.cups.org/doc/options.html) and the lp manpage for further details.

As of SumatraPDF: I'm afraid that would exceed the amount of work I'm able to put in. I wouldn't be able to test it and the docs are quite limited. Only thing I could find, doing a quick search in the docs would be `-print-settings "fit"`, which I assume should be similar to the `fit-to-page` CUPS option. But again: I'm afraid somebody with access to a Windows machine will have to implement and test this.

**Comment by ekoeryanto** [comment]
quick shoot. think to add an option, may be `options.args` so we can maximize both lp and sumatra commands

**Comment by artiebits** [comment]
@ekoeryanto @dVelopment might be a solution but let me think a bit more about the problem before we do any changes. Sorry for the delays in response, and I really appreciate your contributions!

**Comment by ekoeryanto** [comment]
also, i think `execFile` is better than `exec`

**Comment by artiebits** [comment]
@ekoeryanto thanks for suggestions! I would be glad if you open a PR with replacing `exec` with `execFile`. It should be faster and easier to review and merge it then finding a solution for `options`.

**Comment by lahvey** [comment]
I think we need unified options for both unix and windows. The following options is copied from Sumatra PDF for windows: https://www.sumatrapdfreader.org/docs/Command-line-arguments.html

> -print-settings <settings-list>
> Used in combination with -print-to and -print-to-default . Allows to tweak some of the printing related settings without using the Print dialog. The settings-list is a comma separated list of page ranges and advanced options such as
> even or odd
> noscale , shrink and fit
> color or monochrome
> duplex , duplexshort , duplexlong and simplex
> bin=<num or name> : select tray to print to
> E.g. -print-settings "1-3,5,10-8,odd,fit,bin=2" prints pages 1, 3, 5, 9 (i.e. the odd pages from the ranges 1-3, 5-5 and 10-8) and scales them so that they fit into the printable area of the paper.
> -print-settings "3x" : prints the document 3 times

**Comment by dVelopment** [comment]
I don’t think trying to anticipate all possible options is a good idea. I would leave it up to the developers who use this library to be able to provide the options they need. If they don‘t know the environment when writing, they can easily create a dynamic options object on runtime.

**Pull request closed without merging**